bin/controller.py

```diff
@@ -157,8 +164,35 @@ def get_html(snippetid, ext=None):
         ext=ext,
         snippetid=snippetid,
         parentid=snippet.parentid,
+        token=bt.request.query.token,
     )
 
+@bt.route('/<snippetid>', method='DELETE')
+@bt.route('/<snippetid>.<ext>', method='DELETE')
+def delete_snippet(snippetid, ext=None):
+    """
+    Delete a snippet
+
+    :param Authorization: (header) required "Authorization: Token <ADMIN_TOKEN>" the "admin" token
+
+    :raises HTTPError: code 400 when the Authorization (token) is missing
+    :raises HTTPError: code 404 when the snippet is not found
+    :raises HTTPError: code 401 when the token is incorrect
+    """
+    auth = bt.request.get_header('Authorization', '').split(None, 1)
+    if len(auth) != 2 or auth[0] != 'Token':
+        raise bt.HTTPError(400, "Token is missing")
+
+    try:
+        snippet = models.Snippet.get_by_id(snippetid)
+    except KeyError:
+        raise bt.HTTPError(404, "Snippet not found")
+
+    if not (snippet.token and secrets.compare_digest(snippet.token, auth[1])):
+        raise bt.HTTPError(401, "Unauthorized")
+
+    snippet.delete()
```

Comment:
On peut ajouter un peu de log ici ?
```suggestion
    snippet.delete()
    logger.info("Snippet %s deleted by user", snippetid)
```